lib/galaxy/schema/schema.py

```diff
@@ -699,11 +697,14 @@ class HDAInaccessible(HDACommon):
     state: DatasetStateField
 
 
-HdaLddaField = Field(
-    DatasetSourceType.hda,
-    title="HDA or LDDA",
-    description="Whether this dataset belongs to a history (HDA) or a library (LDDA).",
-)
+HdaLddaField = Annotated[
+    DataItemSourceType,
```

Comment:
I think what we want here is `DatasetSourceType` :thinking: 
```suggestion
    DatasetSourceType,
```